examples/stable_fluid.py

```diff
@@ -62,7 +62,7 @@ def bilerp(vf, p):
     u, v = p
     s, t = u - 0.5, v - 0.5
     # floor
-    iu, iv = int(s), int(t)
+    iu, iv = ti.floor(s), ti.floor(t)
```

Comment:
```py
>>> import math
>>> math.floor(-1.1)
-2
>>> int(-1.1)
-1
```
So `int` is quite different from `floor` when things turn negative. The same to Taichi-scope.
This causes the -x and -y side having funny effects when non-zero velocity is encountered.
Hi, dear @Eydcao:
Did you assumed `int` to be exactly same with `ti.floor`, return the largest integer below the input argument?
It warns that we should never do assumptions when debugging. Hope we all have fun and learnt a lot from this vivid lesson :)

Reply:
Interesting!

but in the old code, we used `int()` instead of `math.floor()`, so in python these two are the same thing?

Reply:
Both in Python and Taichi, they are different thing :)

Reply:
Great! This one is good to merge.

I have got the idea why `int()` is irrigating here as it leads to the wrong coefficients in `lerp`. So when I fixed these two lines in #1764.

**The strange boundary artifact is gone, great!**

The new `bfecc` works perfec with `jacobi_single` now, **but still has problems with `jacobi_dual` and `mgpcg`**.